internal/backup_push_handler.go

```diff
@@ -192,6 +192,13 @@ func HandleBackupPush(uploader *Uploader, archiveDirectory string, isPermanent b
 		tracelog.ErrorLogger.Printf("Failed to upload sentinel file for backup: %s", backupName)
 		tracelog.ErrorLogger.FatalError(err)
 	}
+
+	if isPermanent {
```

Comment:
Does it work properly in the case when the current backup is full?

Reply:
actually it doesn't, let me fix that...